core/src/main/scala/spire/math/Sorting.scala

```diff
@@ -36,7 +36,7 @@ object InsertionSort extends Sort {
     * @param end the index of the last element, exclusive, to be sorted
     * @tparam A a member of the type class `Order`
     */
-  final def sort[@sp A](data:Array[A], start:Int, end:Int)(implicit o:Order[A], ct:ClassTag[A]): Unit = {
+  final def sort[@sp A: ClassTag](data:Array[A], start:Int, end:Int)(implicit o:Order[A]): Unit = {
```

Comment:
could be combined into `: ClassTag : Order`

Reply:
The warning here is because the variable is not in use, It maybe better to unify the style and use always all as implicit vals or in the type param

Reply:
If the variable is not in use, couldn't we get rid of the context bound too?

Reply:
The warning complains that the variable `ct` is unused but the implicit ClassTag is used so we need the context bound.

I tried not to touch too much the implicits because in some cases they are not really used in the function but to constraints the callers. The most obvious one is when a `NoImplicit` is required

Reply:
That's weird, but in that case, I guess we can keep it as is.